zcash_proofs/src/downloadreader.rs

```diff
@@ -0,0 +1,79 @@
+//! [`io::Read`] implementations for [`minreq`].
+
+use std::io;
+
+/// A wrapper that implements [`io::Read`] on a [`minreq::ResponseLazy`].
+pub enum ResponseLazyReader {
+    Request(minreq::Request),
+    Response(minreq::ResponseLazy),
+    Complete(Result<(), String>),
+}
+
+impl From<minreq::Request> for ResponseLazyReader {
+    fn from(request: minreq::Request) -> ResponseLazyReader {
+        ResponseLazyReader::Request(request)
+    }
+}
+
+impl From<minreq::ResponseLazy> for ResponseLazyReader {
+    fn from(response: minreq::ResponseLazy) -> ResponseLazyReader {
+        ResponseLazyReader::Response(response)
+    }
+}
+
+impl io::Read for ResponseLazyReader {
+    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
+        use ResponseLazyReader::*;
```

Comment:
nit: we prefer to use individual imports rather than wildcard imports

Reply:
Fixed in 9c331797